File: src/Converter/Input/TACT.pas
wbRecord(TACT, 'Talking Activator',
  wbFlags(wbRecordFlagsFlags, wbFlagsList([
    {0x00000200}  9, 'Hidden From Local Map',
    {0x00010000} 16, 'Random Anim Start',
    {0x00020000} 17, 'Radio Station'
  ]), [17]), [
  wbEDID,
  wbVMAD,
  wbOBNDReq,
  wbFULL,
  wbMODL,
  wbDEST,
  wbKSIZ,
  wbKWDAs,
  wbUnknown(PNAM, cpIgnore, True),
  wbFormIDCk(SNAM, 'Looping Sound', [SNDR]),
  wbUnknown(FNAM, cpIgnore, True),
  wbFormIDCk(VNAM, 'Voice Type', [VTYP])
], False, nil, cpNormal, False, nil, wbKeywordsAfterSet);

File: src/Generator/records/TACT.js
let {
    addDef, record, subrecord, req, def,
    unknown, ckFormId
} = require('../helpers');

module.exports = () => {
    addDef('TACT', record('TACT', 'Talking Activator', {
        flags: {
            9: 'Hidden From Local Map',     // 0x00000200
            16: 'Random Anim Start',        // 0x00010000
            17: 'Radio Station',            // 0x00020000
        },
        elements: [
            def('EDID'),
            def('VMAD'),
            req(def('OBND')),
            def('FULL'),
            def('MODL'),
            def('DEST'),
            def('KSIZ'),
            def('KWDAs'),
            subrecord('PNAM', unknown()),
            subrecord('SNAM', ckFormId('Looping Sound', ['SNDR'])),
            subrecord('FNAM', unknown()),
            subrecord('VNAM', ckFormId('Voice Type', ['VTYP'])),
        ]
    }));
};